### DISCUSSION TITLE
Why is the internal package internal?

### DISCUSSION THREAD

**Discussion by rogchap** [golangci/golangci-lint#2048] [Q&A] [answered]
Is there a reason for the cache code being an internal package, which means it can't be imported outside of the `golangci-lint` module?

I'm creating a `linter.Context` in my project to run some analyzers, but can't instantiate a `pkgCache` because it's an internal package; it's the only thing that is. Without it you get a nil pointer error.

https://github.com/golangci/golangci-lint/blob/fd6fce9e90414fa6dd4753c5dd65c86e21b34058/pkg/lint/linter/context.go#L28

**Comment by alexandear** [comment]
The package in the `internal` directory contains code borrowed from the Go repositories. See #5098

**Comment by ldez** [comment, marked as answer]
golangci-lint is not designed to be used as a library.

We don't want to do that, because we want to keep the freedom of breaking the code.

So don't use it as a library.
